packages/http-crawler/src/internals/file-download.ts

```diff
@@ -0,0 +1,224 @@
+> = InternalHttpHook<FileDownloadCrawlingContext<UserData, JSONData>>;
+
+export interface FileDownloadCrawlingContext<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+> extends InternalHttpCrawlingContext<UserData, JSONData, FileDownload> {}
+
+export type FileDownloadRequestHandler<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+> = RequestHandler<FileDownloadCrawlingContext<UserData, JSONData>>;
+
+/**
+ * Provides a framework for the parallel file download using plain HTTP requests. The URLs to download are fed either from a static list of URLs or can be added dynamically from another source.
+ *
+ * Since `FileDownload` uses raw HTTP requests to download the files, it is very fast and efficient on data bandwidth.
+ * However, it doesn't parse the content - if you need to e.g. extract data from the downloaded files,
+ * you might need to use [CheerioCrawler](https://crawlee.dev/api/cheerio-crawler/class/CheerioCrawler),
+ * [PuppeteerCrawler](https://crawlee.dev/api/puppeteer-crawler/class/PuppeteerCrawler) or [PlaywrightCrawler](https://crawlee.dev/api/playwright-crawler/class/PlaywrightCrawler) instead.
+ *
+ * `FileCrawler` downloads each URL using a plain HTTP request and then invokes the user-provided [FileCrawlerOptions.requestHandler](https://crawlee.dev/api/file-crawler/interface/FileCrawlerOptions#requestHandler) where the user can specify what to do with the downloaded data.
+ *
+ * The source URLs are represented using [Request](https://crawlee.dev/api/core/class/Request) objects that are fed from [RequestList](https://crawlee.dev/api/core/class/RequestList) or [RequestQueue](https://crawlee.dev/api/core/class/RequestQueue) instances provided by the [FileCrawlerOptions.requestList](https://crawlee.dev/api/file-crawler/interface/FileCrawlerOptions#requestList) or [FileCrawlerOptions.requestQueue](https://crawlee.dev/api/file-crawler/interface/FileCrawlerOptions#requestQueue) constructor options, respectively.
+ *
+ * If both [FileCrawlerOptions.requestList](https://crawlee.dev/api/file-crawler/interface/FileCrawlerOptions#requestList) and [FileCrawlerOptions.requestQueue](https://crawlee.dev/api/file-crawler/interface/FileCrawlerOptions#requestQueue) are used, the instance first processes URLs from the [RequestList](https://crawlee.dev/api/core/class/RequestList) and automatically enqueues all of them to [RequestQueue](https://crawlee.dev/api/core/class/RequestQueue) before it starts their processing. This ensures that a single URL is not crawled multiple times.
```

Comment:
Why so? The last time I checked (last week), the Delivery team loves a good `RequestList` (for whatever reason). 

Plus, this behaviour is coming from `BasicCrawler` ([snippet here](https://github.com/apify/crawlee/blob/6da86a85d848cd1cf860a28e5f077b8b14cdb213/packages/basic-crawler/src/internals/basic-crawler.ts#L1096)), so all our crawlers are doing this right now. I'm not here to gatekeep this, just curious about your motivation to remove this :)

Reply:
> for whatever reason

The cost is indeed a valid reason, since with RL there is no cost for the RQ writes (and they can be indeed costly for a high number of requests, e.g. in [this run](https://console.apify.com/actors/Tcx7qJL5NP2PvmZ5c/runs/DoZgMmgZTKHaVSLdX#log) the RQ writes are about 70% of the cost). I will recheck this with RL in place to see what is the actual cost difference.

Reply:
> was not really necessary for this use case

Right, but it's free (inherited from `BasicCrawler`). I copied (and edited) the JSDoc from `HttpCrawler`, so it may be too descriptive in some places, happy to get some feedback for that too. This part is mentioned in all the crawlers we have, though.

Reply:
> The cost is indeed a valid reason, since with RL there is no cost for the RQ writes (and they can be indeed costly for a high number of requests, e.g. in [this run](https://console.apify.com/actors/Tcx7qJL5NP2PvmZ5c/runs/DoZgMmgZTKHaVSLdX#log) the RQ writes are about 70% of the cost). I will recheck this with RL in place to see what is the actual cost difference.

Btw [here](https://console.apify.com/actors/Tcx7qJL5NP2PvmZ5c/runs/Dod3V820KCknZTT5F#log) is the same actor, but with RL in place. The cost is ~5$ while with RQ it's 26$, this is surely a good enough reason to keep this abstraction (unfortunately, lol).